homeassistant/components/icloud/account.py

```diff
@@ -110,9 +115,36 @@ def setup(self) -> None:
                 self._icloud_dir.path,
                 with_family=self._with_family,
             )
-        except PyiCloudFailedLoginException as error:
+        except PyiCloudFailedLoginException:
             self.api = None
-            _LOGGER.error("Error logging into iCloud Service: %s", error)
+            # Login failed which means credentials need to be updated.
+            username = self._config_entry.data[CONF_USERNAME]
+            message = (
+                f"Your password for '{username}' is no longer working. Go to the "
+                "Integrations menu and click on Configure on the discovered Apple "
+                "iCloud card to login again."
+            )
+            _LOGGER.error(message)
+            self.hass.components.persistent_notification.async_create(
```

Comment:
you're callin an async function from a sync context.

Reply:
I removed this per the other comment